src/tools/tidy/src/features/version.rs

```diff
@@ -0,0 +1,66 @@
+use std::str::FromStr;
+use std::num::ParseIntError;
+use std::fmt;
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct Version {
+    parts: Vec<u32>,
```

Comment:
Did you consider using `[u32; 3]` here?

Reply:
Alternatively, try `ArrayVec` if we want to keep the possibility of having fewer items.

Reply:
All the codebase except `src/libstd/sys/redox/ext/net.rs` used 3-part versions. I have converted the file to use `1.29.0` instead of `1.29`, and use `[u32; 3]` now.

Reply:
Excellent!